imagetool/src/main/resources/docker-files/install-middleware.mustache

```diff
@@ -35,9 +35,12 @@ USER {{userid}}
 RUN echo "INSTALLING MIDDLEWARE" \
 {{#installPackages}}
     && echo "INSTALLING {{type}}" \
-    && {{#isZip}}unzip -q {{{tempDir}}}/{{installerFilename}} "*.jar" -d {{{tempDir}}} &&{{/isZip}} \
-    {{{java_home}}}/bin/java -Xmx1024m -jar {{{tempDir}}}/{{jarName}} -silent ORACLE_HOME={{{oracle_home}}} \
-    -responseFile {{{tempDir}}}/{{responseFile.name}} -invPtrLoc {{inv_loc}}/oraInst.loc -ignoreSysPrereqs -force -novalidation \
+    && {{#isZip}}unzip -q {{{tempDir}}}/{{installerFilename}} "*.[jb][ai][rn]" -d {{{tempDir}}} &&{{/isZip}} \
+    {{^isBin}}{{{java_home}}}/bin/java -Xmx1024m -jar {{{tempDir}}}/{{jarName}} -silent ORACLE_HOME={{{oracle_home}}} \
```

Comment:
The line 38 starts with && right? If installer file is not a zip, then && at the start of line 38 will connect the echo command with the installer command. If && added at the beginning of line 39 then there will be consecutive &&

```
Step 24/40 : RUN echo "INSTALLING MIDDLEWARE"     && echo "INSTALLING fmw"     &&  /u01/jdk/bin/java -Xmx1024m -jar /tmp/imagetool/fmw_12.2.1.4.0_infrastructure.jar -silent ORACLE_HOME=/u01/oracle     -responseFile /tmp/imagetool/fmw.rsp -invPtrLoc /u01/oracle/oraInst.loc -ignoreSysPrereqs -force -novalidation       && echo "INSTALLING oid"     && unzip -q /tmp/imagetool/fmw_12.2.1.4.0_oid_linux64_Disk1_1of1.zip "*.[jb][ai][rn]" -d /tmp/imagetool &&  chmod +x /tmp/imagetool/fmw_12.2.1.4.0_oid_linux64.bin &&     /tmp/imagetool/fmw_12.2.1.4.0_oid_linux64.bin -force -ignoreSysPrereqs -silent -responseFile /tmp/imagetool/oid.rsp     -invPtrLoc /u01/oracle/oraInst.loc ORACLE_HOME=/u01/oracle -jreLoc /u01/jdk  && test $? -eq 0 && chmod -R g+r /u01/oracle || (grep -vh "NOTIFICATION" /tmp/OraInstall*/install*.log && exit 1)
 ---> Running in e0a49691d82e
```
In case of both fmw installer not a zip file and oid installer a zip file, the echo statement is connected to installer command with &&

Reply:
Ok, sorry, you are correct.  The && at the start of line 38 is outside the "is this a zip".